src/features/journeys/store.ts

```diff
@@ -55,6 +55,13 @@ const journeysSlice = createSlice({
         }
       }
     },
+    invalidateJourneyInstanceMilestones: (
+      state,
+      action: PayloadAction<number>
+    ) => {
+      const instanceId = action.payload;
+      state.milestonesByInstanceId[instanceId].isStale = true;
```

Comment:
I actually do think that this could be added to the existing `invalidateJourneyInstance` action. If the journey instance has changed, I think it's fair to also reload the state of the milestones, which is a cheap operation.